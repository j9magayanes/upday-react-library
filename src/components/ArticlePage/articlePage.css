

h1 {
    font-family: Roboto Bold,sans-serif;
    margin: 0;
    padding: 0;
    color: black
}

.storybook-title {
    color:black
}

.storybook-title--web {
    color: black;
  }

  .storybook-title--huawei {
    color: black;
  }

  .articleDetail {

    background-color: red !important;
  }

  .readMore {
    margin: 100px;
  }