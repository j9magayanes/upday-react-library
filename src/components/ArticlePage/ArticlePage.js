import React, { Children } from 'react';
import './articlePage.css';
import { Ad } from '../Ad/Ad';
import { CategoryDetail } from '../CategoryDetail/CategoryDetail';
import { Title } from '../Title/Title';
import { Source } from '../Source/Source';
import { RelativeTimestamp } from '../RelativeTimestamp/RelativeTimestamp';
import { FullText } from '../FullText/FullText';
import { HeroImage } from '../HeroImage/HeroImage';
import { ImageCredits } from '../ImageCredits/ImageCredits'
import { ReadMore } from '../ReadMore/ReadMore'

export const ArticlePage = ({
  primary,
  backgroundColor,
  size,
  label,
  title,
  args,
}) => {
  return (
    <>
      <Ad args={args}></Ad>
      <CategoryDetail args={args}></CategoryDetail>
      <Title args={args}/>
      <Source args={args}/>
      <RelativeTimestamp args={args}/>
      <HeroImage args={args}></HeroImage>
      <ImageCredits args={args}></ImageCredits>
      <FullText  args={args}></FullText>
      <ReadMore args={args}></ReadMore>
      <Ad args={args}></Ad>
    </>
  );
};
