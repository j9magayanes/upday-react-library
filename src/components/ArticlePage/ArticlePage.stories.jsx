import React, { Children } from 'react';
import './articlePage.css';
import { ArticlePage } from './ArticlePage';

export default {
    title: 'Form/ArticlePage',
    component: ArticlePage,
  };
  
  const Template = (args) =><ArticlePage   {...args} />;
  
  export const Meinestadt = Template.bind({});
  Meinestadt.args = {
    label: 'meinestadt',
    title: 'Mehr als 16 Monate nach Beginn des russischen Angriffskriegs hat der ukrainische Präsident Wolodymyr Selenskyj die Forderung nach einer baldigen konkreten EU-Beitrittsperspektive für sein Land erneuert. „Die Ukraine hat Einfluss auf die Stärke Europas. Das ist ein Fakt“, sagte Selenskyj am Donnerstag in seiner abendlichen Videoansprache. „Und dieses Jahr ist es an der Zeit, diesen und andere Fakten zu nutzen, um die Einheit in Europa zu stärken – angefangen beim Start von Verhandlungen über die EU-Mitgliedschaft der Ukraine.“ Unterdessen reiste der frühere US-Vizepräsident Mike Pence in die Ukraine. Von Selenskyj in Kiew empfangen wurde zudem die schwedische Umweltaktivistin Greta Thunberg.',
    id: 'meinestadt',
    size: 'meinestadt',
    appName: 'meinestadt',
    class: 'ad',
    src: 'https://assets-static.invideo.io/images/large/Facebook_black_Friday_ads_1f13ff37f9.webp',
    args:{   src: 'https://img.yana.asideas.de/image/upload/w_700,h_394,c_fill,f_webp,q_auto:eco/v1689066061/cms-cropped/hesgy7puvocvswz5pxw6.jpg', appName: 'meinestadt', category: 'Politics', title: 'Selenskyj fordert EU-Beitrittsverhandlungen', source: 'Meinestadt', detail: '> 1 day',  fulltext: 'Mehr als 16 Monate nach Beginn des russischen Angriffskriegs hat der ukrainische Präsident Wolodymyr Selenskyj die Forderung nach einer baldigen konkreten EU-Beitrittsperspektive für sein Land erneuert. „Die Ukraine hat Einfluss auf die Stärke Europas. Das ist ein Fakt“, sagte Selenskyj am Donnerstag in seiner abendlichen Videoansprache. „Und dieses Jahr ist es an der Zeit, diesen und andere Fakten zu nutzen, um die Einheit in Europa zu stärken – angefangen beim Start von Verhandlungen über die EU-Mitgliedschaft der Ukraine.“',  text: 'Read More',  id: 'meinestadt', class: 'hero'}

  };
  
  export const Web = Template.bind({});
  Web.args = {
    title: 'Mehr als 16 Monate nach Beginn des russischen Angriffskriegs hat der ukrainische Präsident Wolodymyr Selenskyj die Forderung nach einer baldigen konkreten EU-Beitrittsperspektive für sein Land erneuert. „Die Ukraine hat Einfluss auf die Stärke Europas. Das ist ein Fakt“, sagte Selenskyj am Donnerstag in seiner abendlichen Videoansprache. „Und dieses Jahr ist es an der Zeit, diesen und andere Fakten zu nutzen, um die Einheit in Europa zu stärken – angefangen beim Start von Verhandlungen über die EU-Mitgliedschaft der Ukraine.“ Unterdessen reiste der frühere US-Vizepräsident Mike Pence in die Ukraine. Von Selenskyj in Kiew empfangen wurde zudem die schwedische Umweltaktivistin Greta Thunberg.',
    size: 'web',
    class: 'ad',
    appName: 'web',
    id: 'web',
    src: 'https://img.yana.asideas.de/image/upload/w_700,h_394,c_fill,f_webp,q_auto:eco/v1689066061/cms-cropped/hesgy7puvocvswz5pxw6.jpg',
    args:{   src: 'https://img.yana.asideas.de/image/upload/w_700,h_394,c_fill,f_webp,q_auto:eco/v1689066061/cms-cropped/hesgy7puvocvswz5pxw6.jpg', appName: 'web', category: 'Business', title: 'Selenskyj fordert EU-Beitrittsverhandlungen', source: 'Upday Web', detail: '> 1 day',  fulltext: 'Mehr als 16 Monate nach Beginn des russischen Angriffskriegs hat der ukrainische Präsident Wolodymyr Selenskyj die Forderung nach einer baldigen konkreten EU-Beitrittsperspektive für sein Land erneuert. „Die Ukraine hat Einfluss auf die Stärke Europas. Das ist ein Fakt“, sagte Selenskyj am Donnerstag in seiner abendlichen Videoansprache. „Und dieses Jahr ist es an der Zeit, diesen und andere Fakten zu nutzen, um die Einheit in Europa zu stärken – angefangen beim Start von Verhandlungen über die EU-Mitgliedschaft der Ukraine.“',  text: 'Read More', id: 'web', class: 'hero' }

  };
  
 