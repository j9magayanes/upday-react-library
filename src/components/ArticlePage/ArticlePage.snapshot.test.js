import React from 'react';
import renderer from 'react-test-renderer';
import { ArticlePage } from './ArticlePage';

test('renders ArticlePage component correctly', () => {
  const props = {
    primary: true,
    backgroundColor: 'your-background-color',
    size: 'your-size',
    label: 'your-label',
    title: 'your-title',
    appName: 'your-appName',
    src: 'your-image-src',
    args: { /* your args object */ },
  };

  const component = renderer.create(<ArticlePage {...props} />);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
