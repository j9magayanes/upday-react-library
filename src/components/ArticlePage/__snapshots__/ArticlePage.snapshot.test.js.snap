// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ArticlePage component correctly 1`] = `
Array [
  <figure>
    <ad-wrapper
      class="ad-block grow"
    >
      <h3>
        Story continues below advertisement
      </h3>
      <image-component
        args={
          Object {
            "imageRef": null,
          }
        }
        src="your-image-src"
      />
    </ad-wrapper>
  </figure>,
  <category-detail
    id="category-detail"
  />,
  <h1
    className="title"
  />,
  <h1
    className="source"
  />,
  <span
    className="timestamp"
  />,
  <figure>
    <img />
  </figure>,
  <span
    className="imageCredits"
  >
    Baden-Württemberg, Ulm: Der Angeklagte wird in Hand- und Fußschellen in den Gerichtssaal geführt. Foto: Felix Kästle/dpa
  </span>,
  <article-detail
    data-testid="articleDetail"
  />,
  <span
    id="storybook-read-more-undefined"
  />,
  <figure>
    <ad-wrapper
      class="ad-block grow"
    >
      <h3>
        Story continues below advertisement
      </h3>
      <image-component
        args={
          Object {
            "imageRef": null,
          }
        }
        src="your-image-src"
      />
    </ad-wrapper>
  </figure>,
]
`;
